x/nexus/keeper/lockable_asset.go

```diff
@@ -174,38 +167,55 @@ func mint(ctx sdk.Context, bank types.BankKeeper, toAddr sdk.AccAddress, coin sd
 	return nil
 }
 
-func getCoinType(ctx sdk.Context, nexus types.Nexus, ibc types.IBCKeeper, denom string) (types.CoinType, error) {
+// normalizeCoin normalizes the given denom to the corresponding registered asset and returns the coin type
+func normalizeCoin(ctx sdk.Context, nexus types.Nexus, ibc types.IBCKeeper, coin sdk.Coin) (types.CoinType, sdk.Coin, error) {
```

Comment:
```suggestion
func toAssetAndType(ctx sdk.Context, nexus types.Nexus, ibc types.IBCKeeper, coin sdk.Coin) (types.CoinType, sdk.Coin, error) {
```
Hate the word `normalize` as it is meaningless.